### PR TITLE
Added salabim and salabim-animation.

### PR DESCRIPTION
**Pull request by moorepants**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by moorepants** [comment]
This package can run with only Python but many of its interesting features are via the animations which require a set of optional dependencies. I decided to make two packages: salabim (barebones no deps) and salabim-animation that requires just the packages needed for the 2D and 3D animations. There are also other optional dependencies I put into run_constrained. Lastly, pyglet is an optional dependency but likely requires a very strict version pin. I didnt' want that run_constrained strict pin to limit installing salabim in a broad set of envirionments, especially if the user isnt' using pyglet, so I left the pin off. I put the pinless packages in run_constrained just to have a log of what packages salabim optionally depends on. salabim does not report and minimum versions for the dependencies in its docs.

**Comment by moorepants** [comment]
@conda-forge/help-python this is ready for review, thanks.

**Comment by moorepants** [comment]
ping @conda-forge/help-python

File: recipes/salabim/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+test:
+  imports:
+    - salabim
+  commands:
+    - pip check
+  requires:
+    - pip
+
+outputs:
+
```
Comment (isuruf) [inline recipes/salabim/meta.yaml:48]:
```suggestion
  - name: salabim
```

File: recipes/salabim/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        - python >=3.6
+      run:
+        - python >=3.6
+        - salabim
+        # all of these are optional dependencies but are needed for the
+        # animations
+        - numpy
+        - opencv
+        - pillow
+        - pyopengl
```
Comment (isuruf) [inline recipes/salabim/meta.yaml:62]: Add a simple test for this pacakge (even a import salabim is fine) so that the run environment is created at least.

**Comment by moorepants** [comment]
@conda-forge-admin, please rerender

File: recipes/salabim/meta.yaml (hunk 2 above)
Comment (moorepants) [reply recipes/salabim/meta.yaml:62]: Done

**Comment by moorepants** [comment]
ping @conda-forge/help-python can this be merged? Any other things I need to fix?

File: recipes/salabim/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    - pillow
+    # salabim docs say this stricth pin is needed, but worried that this will
+    # be problematic in environment solves
+    - pyglet  # ==1.5.27
+    - pyopengl
+    - pysimplegui
+    - python-dateutil
+    - scipy
+    # no feedstock, but may be added in future
+    # - pywavefront
```
Comment (xhochy) [inline recipes/salabim/meta.yaml:37]: Why does pip check work if this isn't available? Is it an optional dependency?

Comment (moorepants) [reply recipes/salabim/meta.yaml:37]: They all optional dependencies that are not specified programmatically upstream. They are only listed in the documentation.

**Comment by moorepants** [comment]
Thanks!

**Pull request merged**